File: Data Structures & Algorithms/plus-one/submission-0.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int c = 1, temp;
        vector<int> ans;
        for(int i = digits.size()-1 ; i>= 0 ; i--){
            temp = c + digits[i];
            c = temp > 9 ? 1 : 0;
            ans.push_back(temp%10); 
        }
        if(c)
            ans.push_back(1);
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: Data Structures & Algorithms/task-scheduling/submission-0.java
class Solution {
    class Pair {
        int cnt;
        char ch;
        Pair(char ch, int cnt) {
            this.ch = ch;
            this.cnt = cnt;
        }
    }
    public int leastInterval(char[] tasks, int n) {
        Map<Character, Integer> map = new HashMap<>();
        for (char c : tasks) {
            map.put(c, map.getOrDefault(c, 0) + 1);
        }

        PriorityQueue<Pair> pq = new PriorityQueue<>((a, b) -> b.cnt - a.cnt);
        Queue<Pair> pqcd = new LinkedList<>();

        for (Map.Entry<Character, Integer> ele : map.entrySet()) {
            pq.offer(new Pair(ele.getKey(), ele.getValue()));
        }

        int ans = 0;
        while (map.size() > 0) {
            ans++;
            if ((pqcd.size() > 0) && pqcd.peek().cnt < ans) {
                pq.offer(new Pair(pqcd.peek().ch, map.get(pqcd.peek().ch)));
                pqcd.poll();
            }
            if (pq.isEmpty())
                continue;
            Pair temp = pq.poll();

            int cnt = map.get(temp.ch);
            if (cnt - 1 == 0) {
                map.remove(temp.ch);
            } else {
                map.put(temp.ch, cnt - 1);
                pqcd.offer(new Pair(temp.ch, ans + n));
            }
        }
        return ans;
    }
}
